spot-contracts/contracts/vaults/RolloverVault.sol

```diff
@@ -315,6 +338,99 @@ contract RolloverVault is
         return redemptions;
     }
 
+    /// @notice Swaps the bond tranche slices from the user into the underlying collateral for a fee,
+    ///         given that the vault has the remaining slices for immature redemption.
+    /// @dev When a user has some tranche slices from a given bond, and the vault has the remaining slices,
+    ///      this method allows the user and the vault to "meld" their assets together to redeem the underlying collateral
+    ///      from the bond. The user's share of the underlying is returned back and the vault charges
+    ///      the user a fee for providing liquidity.
+    /// @param bond The bond whose tranches are to be swapped.
+    /// @param trancheAmtsIn A list of amounts of each bond tranche the user is to deposit.
+    ///                      When the user does not have a particular slice, its amount in the list is to be set to zero.
+    /// @return The amount of underlying tokens returned.
+    function meld(IBondController bond, uint256[] memory trancheAmtsIn)
+        external
+        nonReentrant
+        whenNotPaused
+        returns (uint256)
+    {
+        // get bond tranches
+        BondTranches memory bt = bond.getTranches();
+
+        // 1) We check that the bond's underlying collateral token matches the vault's underlying.
+        // 2) We validate that the bond has at least 2 tranches to perform the meld operation.
+        // 3) We validate that the bond hasn't reached maturity as matching only works through immature redemption.
+        // 4) We check if the given bond is NOT malicious, by examining if at least one of the bond's tranches
+        //    is already in the vault from previous deployments. And if the parent bond of the tranche already in the vault
+        //    is the given bond.
+        if (
+            bond.collateralToken() != address(underlying) ||
+            bt.tranches.length <= 1 ||
+            bond.isMature() ||
+            !_isDeployedBond(bond, bt)
```

Comment:
If this check passes... do we need the first two?

Reply:
Right the 3rd check should implicitly cover the first two as well. Since we trust spot to 1) add tranches from only AMPL bonds and 2) be configured with the correct bond issuer.

We can comment 1&2 out as an optimization.